lib/stdlib/test/json_SUITE.erl

```diff
@@ -0,0 +1,507 @@
+%%
+%% %CopyrightBegin%
+%%
+%% Copyright Ericsson AB 2024. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% %CopyrightEnd%
+%% % @format
+%%
+%%%----------------------------------------------------------------
+%%% Purpose: Test suite for the 'json' module.
+%%%-----------------------------------------------------------------
+
+-module(json_SUITE).
+-include_lib("stdlib/include/assert.hrl").
+
+%% Test server specific exports
+-export([all/0, suite/0, groups/0, init_per_group/2, end_per_group/2]).
+
+%% Test cases must be exported.
+-export([
+    test_encode_atom/1,
+    test_encode_integer/1,
+    test_encode_float/1,
+    test_encode_binary/1,
+    test_encode_map/1,
+    test_encode_list/1,
+    test_encode_proplist/1,
+    test_encode_escape_all/1,
+    test_decode_atoms/1,
+    test_decode_numbers/1,
+    test_decode_strings/1,
+    test_decode_arrays/1,
+    test_decode_objects/1,
+    test_decode_whitespace/1,
+    test_decode_api/1,
+    test_json_test_suite/1,
+    counterexamples/1
+]).
+
+-ifdef(PROPER).
+-export([
+    property_string_roundtrip/1,
+    property_integer_roundtrip/1,
+    property_float_roundtrip/1,
+    property_object_roundtrip/1,
+    property_escape_all/1
+]).
+
+-define(PROPER_NO_TRANS, true).
+-define(PROPER_NO_IMPORT_PARSE, true).
+-include_lib("proper/include/proper.hrl").
+-endif.
```

Comment:
AFAIK, property tests are usually put in their own suite, using the `ct_property_test` framework as described [here](https://www.erlang.org/doc/apps/common_test/ct_property_test_chapter). At least all the property tests for other modules in OTP that I know of are organized this way. For example, see [`stdlib/test/base64_property_test_SUITE.erl`](https://github.com/erlang/otp/blob/master/lib/stdlib/test/base64_property_test_SUITE.erl) and the corresponding [`stdlib/test/property_test/base64_prop.erl`](https://github.com/erlang/otp/blob/master/lib/stdlib/test/property_test/base64_prop.erl).